File: Ej 15 (N Armstrong)/Python/Armstrong.py
## DeathbatO

def es_Narcisista(num):
    org = num
    sum = 0
    len_num = len(str(num))

    while num != 0:
        mod = num % 10
        sum += mod**len_num
        num //= 10

    return org == sum

for i in range(5):
    num =  int(input("Por favor ingrese un numero:..."))
    if num > 0:
        if es_Narcisista(num):
            print(f"El numero {num} es Narcisista.")
        else:
            print(f"El numero {num} NO es Narcisita...")
    else:
        print("El numero debe ser mayor a 0")

File: Ej 15 (N Armstrong)/Kotlin/Armstrong.kt
// DeathbatO

import kotlin.math.pow
import java.util.Scanner

fun esNarcisista(num: Int): Boolean{
    var org = num
    var sum = 0
    var len = num.toString().length

    while(org != 0){
        val mod = num % 10
        sum += mod.toDouble().pow(len).toInt()
        org /= 10
    }
    return num == sum
}

fun main(){
    val sc = Scanner(System.`in`)

    repeat(5) {
        print("Por favor, ingrese un número: ")
        val num = sc.nextInt()
        sc.nextLine()

        if (num > 0) {
            if (esNarcisista(num)) {
                println("El número $num es Narcisista.")
            } else {
                println("El número $num NO es Narcisista.")
            }
        } else {
            println("El número debe ser mayor a cero...")
        }
    }
    sc.close()
}